### DISCUSSION TITLE
Sharing Files With --private Option Enabled

### DISCUSSION THREAD

**Discussion by gitsteff** [netblue30/firejail#5590] [Q&A]
Hi.
I am starting firefox like this:

firejail --private firefox

I would like to upload some photos from my mobile phone but can not access it. It is mounted like this:

mtp://HMD_Global_Nokia_8.1_PNXGAM8942600427/

How would I archive this?

Thanks.

**Comment by rusty-snake** [comment]
A. Setup gvfs to work inside the sandbox.
B. Use `firejail --put` or procfs to copy it into the sandbox.
C. Copy/Bind-mount it somewhere else that is shared.

**Reply by gitsteff** [reply]
Thanks for the quick answer, rusty-snake.
Option B. worked.

I am still having trouble to set up gvfs. I found this thread as an possible solution to set it up: 
https://github.com/netblue30/firejail/discussions/4710?sort=top

When using
firejail --debug --private --whitelist=/run/user/1000/gvfs/ --noblacklist=${HOME}/.local/share/gvfs-metadata/ --whitelist=${HOME}/.local/share/gvfs-metadata/ --dbus-user.talk=org.gtk.vfs.* /usr/lib/firefox/firefox %u

it complains like this:
Mounting tmpfs on /run/user/1000, check owner: no
359 352 0:100 / /run/user/1000 rw,nosuid,nodev,relatime - tmpfs tmpfs rw,mode=700,uid=1000,gid=1000,inode64
mountid=359 fsname=/ dir=/run/user/1000 fstype=tmpfs
Error fstat: fs_whitelist.c:156 whitelist_file: Permission denied
Error: proc 12128 cannot sync with peer: unexpected EOF
Peer 12131 unexpectedly exited with status 1

Any help would be very welcome.

Thanks

**Reply by rusty-snake** [reply]
Configure FUSE to mount with `allow_root` or `allow_other` (and understand why this is not the default).

**Reply by gitsteff** [reply]
Huh...
This is kind of a show stopper to me because I am using arch linux with qemu/kvm to redirect the usb port to a Virtual Machine to access the device/files in that VM. Too sad, everything is done automatically when selecting the device over the virt-viewer GUI. So it is not transparent to me what is going on.

Do you have an idea for this use case?

Thanks.
